src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/RazorSemanticTokensLegendEndpoint.cs

```diff
@@ -3,16 +3,29 @@
 
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Interfaces;
+using MediatR;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
 using Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
+    internal class SemanticTokensLegendParams : IRequest<SemanticTokensLegend>, IBaseRequest
+    {
+    }
+
+    [Method(LanguageServerConstants.RazorSemanticTokensLegendEndpoint)]
```

Comment:
Do we still n eed to indicate serial/parallelness?

Reply:
Also, i'd throw these classes/interfaces into their own files despite how simple the legend handler endpoitn is 😄